lab-judy/app/filter/item-search.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+// module.exports = function(){
+//   return function(items, searchTerm){
+//     let fuzzyRegex = generateFuzzyRegex(searchTerm);
+//
+//     let filteredByName = items.filter(item => {
+//       //does the generateFuzzyRegex happen here or next block?
+//       let fuzzyName = fuzzyRegex.test(item.name.toUpperCase());
+//       let fuzzyDesc = fuzzyRegex.test(item.desc.toUpperCase());
+//       return fuzzyName;
+//     });
+//     filtered.filter(item => {
+//       //any chaining of methods?
+//       return item;
+//     });
+//   };
+// };
+
+module.exports = function(){
+  return function(items, searchTerm){
+    let fuzzyRegex = generateFuzzyRegex(searchTerm);
+
+    return items.filter(item => {
+      let fuzzyName = fuzzyRegex.test(item.name.toUpperCase());
+      let fuzzyDesc = fuzzyRegex.test(item.desc.toUpperCase());
+      //run filter on the name, return that, then f/u with filter by desc and return that
+      return fuzzyName || fuzzyDesc;
```

Comment:
Because our generateFuzzyRegex function will return the whole string if there is no input to the filter, we can simply apply the code Duncan used to the gallery array twice, once for each search term input. We can just take in the name and description search terms, filter our galleries by name, save the result of filtering that gallery array, then filter THAT array using the desc. We then return the result of filtering the galleries through both of our filters. Even if one of the filters isn't being utilized, it won't ultimately alter the array due to the way our generateFuzzyRegex function is written.

Here's an example of that in practice: 
```
module.exports = function(){
  return function(items, searchTermName, searchTermDesc){

    let filterArray = [];
    let fuzzyRegex = generateFuzzyRegex(searchTermDesc);
    filterArray =  items.filter(item => {
      return fuzzyRegex.test(item.desc.toUpperCase());
    });

    fuzzyRegex = generateFuzzyRegex(searchTermName);
    filterArray = filterArray.filter(item => {
      return fuzzyRegex.test(item.name.toUpperCase());
    });

    return filterArray;
  };
};
```

You'll have to pass the two arguments to our filter within the filter being applied to the gallery-li element.